integration/common/openlineage/common/provider/dbt/structured_logs.py

```diff
@@ -0,0 +1,508 @@
+            job_namespace=self.dbt_run_metadata.job_namespace,
+            run_facets=run_facets,
+        )
+
+    def _setup_dbt_run_metadata(self, start_event):
+        """
+        The dbt command defines the parent run metadata of
+        all subsequent OL events (NodeStart, NodeFinish ...)
+        """
+        self.dbt_run_metadata = ParentRunMetadata(
+            run_id=start_event.run.runId,
+            job_name=start_event.job.name,
+            job_namespace=start_event.job.namespace,
+        )
+
+    def dbt_version_facet(self) -> DbtVersionRunFacet:
+        return {"dbt_version": DbtVersionRunFacet(version=self.dbt_version)}
+
+    def _get_sql_query_id(self, node_id) -> int:
+        """
+        Not all adapters have the sql id defined in their dbt event.
+        A node is executed by a single thread which means that sql queries of a single node are executed
+        sequentially and their status is also reported sequentially.
+        This function gives us a surrogate query id. It's auto-incremented
+        """
+        query_id = self.node_id_to_sql_query_id[node_id]
+        self.node_id_to_sql_query_id[node_id] += 1
+        return query_id
+
+    def _get_sql_job_name(self, event) -> str:
+        """
+        The name of the sql job is as follows
+        {node_job_name}.sql.{incremental_id}
+        """
+        node_job_name = self._get_job_name(event)
+        node_unique_id = get_node_unique_id(event)
+        query_id = self._get_sql_query_id(node_unique_id)
+        job_name = f"{node_job_name}.sql.{query_id}"
+
+        return job_name
+
+    def _get_job_name(self, event) -> str:
+        """
+        The job name of models, snapshots ...
+        """
+        database = event["data"]["node_info"]["node_relation"]["database"]
+        schema = event["data"]["node_info"]["node_relation"]["schema"]
+        node_unique_id = get_node_unique_id(event)
+        if node_unique_id.startswith("model"):
+            node_id = self.removeprefix(node_unique_id, "model.")
+            suffix = ".build.run" if self.dbt_command == "build" else ""
+        elif node_unique_id.startswith("snapshot"):
+            node_id = self.removeprefix(node_unique_id, "snapshot.")
+            suffix = ".build.snapshot" if self.dbt_command == "build" else ".snapshot"
+        else:
+            node_id = node_unique_id
+            suffix = ".build.run" if self.dbt_command == "build" else ""
+
+        return f"{database}.{schema}.{node_id}{suffix}"
+
+    def _run_dbt_command(self) -> Generator[str, None, None]:
+        """
+        This is a generator, it returns log lines
+        """
+        # log in json and generate the artifacts
+        dbt_command_line = add_command_line_arg(
+            self.dbt_command_line, arg_name="--log-format", arg_value="json"
+        )
+        dbt_command_line = add_or_replace_command_line_option(
+            self.dbt_command_line, option="--write-json", replace_option="--no-write-json"
```

Comment:
```suggestion
            dbt_command_line, option="--write-json", replace_option="--no-write-json"
```
Otherwise it doesn't add `--log-format` from line above.